src/core/unittest/FrameTest.cpp

```diff
@@ -243,6 +243,91 @@ TEST(FrameTest, ReliableResetStreamFrameEncodeDecode)
     ASSERT_EQ(Frame.ReliableSize, DecodedFrame.ReliableSize);
 }
 
+TEST(FrameTest, TestQuicAckTrackerDidHitReorderingThreshold)
+{
+    QUIC_ACK_TRACKER Tracker;
+    uint8_t ReorderingThreshold;
+
+    // Initialize the Tracker and other variables
+
+    QuicAckTrackerInitialize(&Tracker);
+
+    // ReorderingThreshold is 0
+    ReorderingThreshold = 0;
+    QuicRangeAddValue(&Tracker.PacketNumbersToAck, 100);
+    ASSERT_FALSE(QuicAckTrackerDidHitReorderingThreshold(&Tracker, ReorderingThreshold, 100));
+
+    // The number of ranges is less than 2.
+    ReorderingThreshold = 3;
+    QuicRangeAddValue(&Tracker.PacketNumbersToAck, 101);
+    ASSERT_FALSE(QuicAckTrackerDidHitReorderingThreshold(&Tracker, ReorderingThreshold, 101));
+
+    // PacketNumber is not the largest unacked packet number
+    QuicRangeAddValue(&Tracker.PacketNumbersToAck, 104);
+    QuicRangeAddValue(&Tracker.PacketNumbersToAck, 105);
+    ASSERT_FALSE(QuicAckTrackerDidHitReorderingThreshold(&Tracker, ReorderingThreshold, 103));
+
+    // Case 1 
+    QuicAckTrackerReset(&Tracker); 
+    QuicRangeAddValue(&Tracker.PacketNumbersToAck, 0);
+    ASSERT_FALSE(QuicAckTrackerDidHitReorderingThreshold(&Tracker, ReorderingThreshold, 0));
+
+    QuicRangeAddValue(&Tracker.PacketNumbersToAck, 1);
+    ASSERT_FALSE(QuicAckTrackerDidHitReorderingThreshold(&Tracker, ReorderingThreshold, 1));
+
+    QuicRangeAddValue(&Tracker.PacketNumbersToAck, 3);
+    ASSERT_FALSE(QuicAckTrackerDidHitReorderingThreshold(&Tracker, ReorderingThreshold, 3));
+
+    QuicRangeAddValue(&Tracker.PacketNumbersToAck, 4);
+    ASSERT_FALSE(QuicAckTrackerDidHitReorderingThreshold(&Tracker, ReorderingThreshold, 4));
+
+    QuicRangeAddValue(&Tracker.PacketNumbersToAck, 5);
+    ASSERT_TRUE(QuicAckTrackerDidHitReorderingThreshold(&Tracker, ReorderingThreshold, 5));
+    QuicRangeAddValue(&Tracker.PacketNumbersToAck, 2); // 2 is reported missing
```

Comment:
Why are you adding these numbers to `PacketNumbersToAck`. In the spec examples, they're never reported as received.

Reply:
I need a way to declare that we have reported it missing, I am trying to figure out a way else it will keep acknowledging for each subsequent packets as the gap will always be there. Any suggestions?